core/src/main/java/org/infinispan/remoting/responses/PrepareResponse.java

```diff
@@ -29,28 +25,38 @@
  * @author Pedro Ruivo
  * @since 11.0
  */
+@ProtoTypeId(ProtoStreamTypeIds.PREPARE_RESPONSE)
 public class PrepareResponse extends ValidResponse {
 
-   public static final Externalizer EXTERNALIZER = new Externalizer();
-
    private Map<Object, IncrementableEntryVersion> newWriteSkewVersions;
    private Map<Integer, IracMetadata> newIracMetadata;
 
-   public static void writeTo(PrepareResponse response, ObjectOutput output) throws IOException {
-      marshallMap(response.newWriteSkewVersions, output);
-      marshallMap(response.newIracMetadata, DataOutput::writeInt, IracMetadata::writeTo, output);
+   public static PrepareResponse asPrepareResponse(Object rv) {
+      assert rv == null || rv instanceof PrepareResponse;
+      return rv == null ? new PrepareResponse() : (PrepareResponse) rv;
    }
 
-   public static PrepareResponse readFrom(ObjectInput input) throws IOException, ClassNotFoundException {
-      PrepareResponse response = new PrepareResponse();
-      response.newWriteSkewVersions = unmarshallMap(input, HashMap::new);
-      response.newIracMetadata = unmarshallMap(input, DataInput::readInt, IracMetadata::readFrom, HashMap::new);
-      return response;
+   public PrepareResponse() {
    }
 
-   public static PrepareResponse asPrepareResponse(Object rv) {
-      assert rv == null || rv instanceof PrepareResponse;
-      return rv == null ? new PrepareResponse() : (PrepareResponse) rv;
+   @ProtoFactory
+   PrepareResponse(MarshallableObject<?> object, MarshallableCollection<?> collection,
+                   MarshallableMap<?, ?> map, MarshallableArray<?> array, byte[] bytes,
+                   MarshallableMap<Object, IncrementableEntryVersion> newWriteSkewVersions,
+                   MarshallableMap<Integer, IracMetadata> newIracMetadata) {
+      super(null, null, map, null, null);
+      this.newWriteSkewVersions = MarshallableMap.unwrap(newWriteSkewVersions);
+      this.newIracMetadata = MarshallableMap.unwrap(newIracMetadata);
+   }
+
+   @ProtoField(6)
+   MarshallableMap<Object, IncrementableEntryVersion> getNewWriteSkewVersions() {
+      return MarshallableMap.create(newWriteSkewVersions);
+   }
+
+   @ProtoField(7)
+   MarshallableMap<Integer, IracMetadata> getNewIracMetadata() {
```

Comment:
Both `Integer` and `IracMetadata` are supported proto3 Map entities. Wrapping is not required.